bundles/org.openhab.binding.bluetooth/src/test/java/org/openhab/binding/bluetooth/util/RetryFutureTest.java

```diff
@@ -151,10 +152,9 @@ void composeWithRetry1Cancel() throws InterruptedException {
             if (!latch.await(100, TimeUnit.MILLISECONDS)) {
                 fail("Timeout while waiting for latch");
             }
-            Thread.sleep(1);
```

Comment:
The reason for this sleep is to _hopefully_ prevent the very small chance that the retryFuture gets cancelled before the composed future is returned from the function (which in this case would be the very next line of code). Since we know where each thread is in it's processing I think that giving a 1 millisecond delay would be more than sufficient to allow the composed future to return before we attempt to cancel the retryFuture.